sleap/gui/widgets/video.py

```diff
@@ -2063,6 +2068,197 @@ def paint(self, painter, option, widget=None):
         pass
 
 
+class VisibleBoundingBox(QtWidgets.QGraphicsRectItem):
+    """QGraphicsRectItem for user instance bounding boxes.
+
+    This object defines a scalable bounding box that encases an instance and handles
+    the relevant scaling operations. It is instantiated when its respective QtInstance
+    object is instantiated.
+
+    When instantiated, it creates 4 boxes, which are properties of the overall object,
+    on the corners of the overall bounding box. These corner boxes can be dragged to
+    scale the overall bounding box.
+
+    Args:
+        rect: The :class:`QRectF` object which defines the non-scalable bounding box.
+        parent: The :class:`QtInstance` to encompass.
+
+    """
+
+    def __init__(
+        self,
+        rect: QRectF,
+        parent: QtInstance,
+        opacity: float = 0.8,
+        scaling_padding=10.0,
+    ):
+        super().__init__(rect, parent)
+        self.box_width = parent.markerRadius
```

Comment:
Consider the edge case where our `markerRadius` is very large and one or more of the nodes is very close to the corner of the bounding box. Will it ever be the case where the corner boxes occlude the node(s)? Do we need to increase the padding on our visible bounding box to overcome this or is it fine?